app/src/components/dataset/EditDatasetForm.tsx

```diff
@@ -0,0 +1,188 @@
+import React from "react";
+import { Controller, useForm } from "react-hook-form";
+import { graphql, useMutation } from "react-relay";
+
+import {
+  Button,
+  Field,
+  Flex,
+  Form,
+  TextArea,
+  TextField,
+  View,
+} from "@arizeai/components";
+
+import { JSONEditor } from "@phoenix/components/code";
+import { isJSONObjectString } from "@phoenix/utils/jsonUtils";
+
+import {
+  EditDatasetFormMutation,
+  EditDatasetFormMutation$variables,
+} from "./__generated__/EditDatasetFormMutation.graphql";
+import { metadataFieldWrapperCSS } from "./styles";
+
+type PatchDatasetParams = Omit<EditDatasetFormMutation$variables, "datasetId">;
+
+export function EditDatasetForm({
```

Comment:
Very minor nit and can be done as a follow-up but I think if organized well we can make a non relay connected form called DatasetForm and then have Edit and Create variants just be wrappers that have relay stuff and we can put that on pages. It might be worth doing just so we start down this pattern. LMK

Reply:
yup i like it, i was thinking the same when i was making this form, will put it as a follow up if it seems big

Reply:
I think it's probably worth just doing to we can de-couple form logic from relay logic. It seems like the right time to keep things DRY

Reply:
pretty easy refactor so doing it in this pr